ignite/handlers/ema_handler.py

```diff
@@ -15,32 +16,32 @@ class EMAHandler:
     .. math:: \theta_{\text{EMA}, t+1} = (1 - \lambda) \cdot \theta_{\text{EMA}, t} + \lambda \cdot \theta_{t}
 
     where :math:`\theta_{\text{EMA}, t}` and :math:`\theta_{t}` are the EMA weights and online model weights at
-    :math:`t`-th iteration, respectively; :math:`\lambda` is the update momentum. The handler allows for linearly
-    warming up the momentum in the beginning when training process is not stable. Current momentum can be retrieved
+    :math:`t`-th iteration, respectively; :math:`\lambda` is the update momentum. Current momentum can be retrieved
     from ``Engine.state.ema_momentum``.
 
     Args:
           model: the online model for which an EMA model will be computed. If ``model`` is ``DataParallel`` or
               ``DistributedDataParallel``, the EMA smoothing will be applied to ``model.module`` .
           momentum: the update momentum after warmup phase, should be float in range :math:`\left(0, 1 \right)`.
-          momentum_warmup: the initial update momentum during warmup phase, the value should be smaller than
-              ``momentum``. Momentum will linearly increase from this value to ``momentum`` in ``warmup_iters``
-              iterations. If ``None``, no warmup will be performed.
-          warmup_iters: iterations of warmup. If ``None``, no warmup will be performed.
+          momentum_warmup: the initial update momentum during warmup phase. This argument is not used.
+          warmup_iters: iterations of warmup. This argument is not used.
 
     Attributes:
           ema_model: the exponential moving averaged model.
           model: the online model that is tracked by EMAHandler. It is ``model.module`` if ``model`` in
               the initialization method is an instance of ``DistributedDataParallel``.
-          momentum: the update momentum after warmup phase.
-          momentum_warmup: the initial update momentum.
-          warmup_iters: number of warmup iterations.
+          momentum: the update momentum.
 
     Note:
           The EMA model is already in ``eval`` mode. If model in the arguments is an ``nn.Module`` or
           ``DistributedDataParallel``, the EMA model is an ``nn.Module`` and it is on the same device as the online
           model. If the model is an ``nn.DataParallel``, then the EMA model is an ``nn.DataParallel``.
 
+    .. warning::
+        The arguments `momentum_warmup` and `warmup_iters` will be deprecated in the future. In the current version,
+        these two argument have no effect on the momentum. I.e., the momentum will be constant during the training.
+        If users need to change the momentum, please use a ``StateParamScheduler``.
```

Comment:
@sandylaker let's add an explicit example on how to do the momentum warm-up.

cc @fco-dv

Reply:
@vfdev-5 an example could be something like this I suppose, but I agree with your idea of simplification, maybe we could define a common `EMAWarmUp` class somewhere as it seems like users will always use these kind of scheduling ?

```python

    from ignite.handlers import EMAHandler

    model = nn.Linear(2, 1)
    trainer = Engine(lambda e, b: model(b))
    data = torch.rand(100, 2)
    
    class EMAWarmUp:
        def __init__(self, momentum_warmup, warmup_iters, momentum):
            self.momentum_warmup = momentum_warmup
            self.warmup_iters = warmup_iters
            self.momentum = momentum

        def __call__(self, event_index):
            denominator = max(1, self.warmup_iters - 1)
            momentum = self.momentum_warmup + (self.momentum - self.momentum_warmup) * (event_index - 1) / denominator
            return min(self.momentum, momentum)

    ema_state_parameter_scheduler = LambdaStateScheduler(
        param_name="momentum",
        lambda_obj=EMAState(momentum_warmup=0.0001, warmup_iters=10, momentum=0.002),
    )
    trainer = Engine(lambda e, b: model(b))

    ema_handler = EMAHandler(model)
    ema_handler.attach(trainer, name="momentum", event=Events.ITERATION_COMPLETED)

    ema_state_parameter_scheduler.attach(trainer, Events.EPOCH_COMPLETED)
    trainer.add_event_handler(Events.EPOCH_COMPLETED, lambda _: print(trainer.state.momentum))

    trainer.run(data, max_epochs=20)
```

Reply:
I think we can just hide this `EMAWarmUp` and `ema_state_parameter_scheduler` and `ema_state_parameter_scheduler.attach` inside `ema_handler = EMAHandler(model, momentum=0.002, momentum_warmup=0.0001, warmup_iters=10)` : 

```python

class EMAHandler:

    def __init__(self, model, momentum=0.002, momentum_warmup=None, warmup_iters=None):
        # ...
        if momentum_warmup and warmup_iters:
            self.momentum_scheduler = # lazy LambdaStateScheduler creation

    def attach(self, engine, name, ...):
        # ...
        if self.momentum_scheduler is not None:
             # attach self.momentum_scheduler.attach(trainer, ...)
```

Reply:
Looks good !

Reply:
@vfdev-5 and @fco-dv There is a problem when implementing the API as the example above:

We need to pass a state parameter name to `LambdaStateScheduler`. However, we do not know the name until we pass it to `EMAHandler.attach`. 

A dirty fixup would be initializing `self.momentum_scheduler = None` in the `__init__` function. Next, we create the `self.momentum_scheduler = LambdaStateScheduler(...)` in `EMAHandler.attach` function.

Reply:
Yes you're right and thanks for your reply @sandylaker!  for the fix, what you've mentioned could be a solution.  Parameter name may also be passed to the EMAHandler constructor  ? ( like for  `ParamScheduler` and `StateParamScheduler` classes), but I don't which one is the best approach ... maybe @vfdev-5 has another idea.

Reply:
@sandylaker yes, that's right we can not create `self.momentum_scheduler` directly that's why marked as `lazy LambdaStateScheduler creation`.
Here is a detailed implementation I was thinking:
```python
from ignite.handlers import LambdaStateScheduler

class EMAWarmUp:
    def __init__(self, momentum, momentum_warmup, warmup_iters):
        self.momentum = momentum        
        self.momentum_warmup = momentum_warmup
        self.warmup_iters = warmup_iters

    def __call__(self, event_index):
        denominator = max(1, self.warmup_iters - 1)
        momentum = self.momentum_warmup + (self.momentum - self.momentum_warmup) * (event_index - 1) / denominator
        return min(self.momentum, momentum)

class EMAHandler:

    def __init__(self, model, momentum=0.002, momentum_warmup=None, warmup_iters=None):
        # ...
        if momentum_warmup and warmup_iters:
            self.momentum_scheduler = None
            self._momentum_lambda_obj = EMAWarmUp(
                momentum=momentum, momentum_warmup=momentum_warmup, warmup_iters=warmup_iters
            )

    def attach(self, engine, name, event):
        # ...
        if self._momentum_lambda_obj is not None:
            self.momentum_scheduler = LambdaStateScheduler(
                param_name=name,
                lambda_obj=self._momentum_lambda_obj
            )            
            self.momentum_scheduler.attach(engine, Events.ITERATION_STARTED)
        # ...
```

What do you think ?

Reply:
@sandylaker @fco-dv do you think we can implement above solution in this PR ?

Reply:
@vfdev-5 Sorry for the delay. 
Yes, I actually have the same idea as you. I will update this PR in this week.